File: pinyougou-manager-web/src/main/webapp/js/service/itemCatService.js
//这是分类的服务层
app.service('itemCatService',function ($http) {
    //查找到所有的最顶端的分类
    this.findByCondition=function (page,rows) {
        return $http.get("../itemCat/findByCondition.do?page="+page+"&rows="+rows);
    }

    //查找到所有的父类的id
    this.findByParentId=function (parentId) {
        return $http.get("../itemCat/findByParentId.do?parentId="+parentId);
    }

    //增加分类的操作
    this.add=function (entity) {
        entity.typeId = entity.typeId.id;
        var methodName="add";
        if(entity.id!=null){//修改
            methodName="update";
        }
        return $http.post("../itemCat/"+methodName+".do",entity);
    }

    //根据id去进行查询
    this.findById=function (id) {
        return $http.get("../itemCat/findById.do?id="+id);
    }

    //根据id进行删除，这是批量删除的操作
    this.dele=function (ids) {
        return $http.get("../itemCat/delete.do?ids="+ids);
    }

    //查找全部的操作
    this.findAll=function () {
        return $http.get("../itemCat/findAll.do");
    }

    this.findOne=function (id) {
        return $http.get("../itemCat/findById.do?id="+id);
    }
})

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.ResponseResult;
import com.pinyougou.pojo.TbItemCat;

import java.util.List;
import java.util.Map;

public interface ItemCatService {
    /** 
    * @Description: 显示所有的分类 
    * @Param: [] 
    * @return: java.util.List<com.pinyougou.pojo.TbItemCat> 
    * @Author: Yin 
    * @Date: 2018/9/4 
    */ 
    List<TbItemCat> findAll();

    /** 
    * @Description: 分页查找到，也包括了条件查询
    * @Param: [page, rows, itemCat] 
    * @return: java.util.Map<java.lang.String,java.lang.Object> 
    * @Author: Yin 
    * @Date: 2018/9/4 
    */ 
    Map<String,Object> findByCondition(int page, int rows);

    /** 
    * @Description: 根据父类的id查到所属的分类
    * @Param: [parentId] 
    * @return: java.util.List<com.pinyougou.pojo.TbItemCat> 
    * @Author: Yin 
    * @Date: 2018/9/4 
    */ 
    List<TbItemCat> findByParentId(Long parentId);

    /** 
    * @Description: 添加分类的操作
    * @Param: [itemCat] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/9/7 
    */ 
    void addItemCat(TbItemCat itemCat);

    /** 
    * @Description: 修改的操作 
    * @Param: [itemCat] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/9/8 
    */ 
    void updateItemCat(TbItemCat itemCat);

    /** 
    * @Description: 根据id去查找到分类的
    * @Param: [id] 
    * @return: com.pinyougou.pojo.TbItemCat 
    * @Author: Yin 
    * @Date: 2018/9/8 
    */ 
    TbItemCat findById(Long id);

    /** 
    * @Description: 批量删除
    * @Param: [ids] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/9/8 
    */
    List<Integer>  deleteItemCat(Long[] ids);
}
